src/NuGet.CommandLine/Commands/RestoreCommand.cs

```diff
@@ -84,7 +92,43 @@ public RestoreCommand()
             }
         }
 
-        private async Task PerformNuGetV3RestoreAsync(string projectPath)
+        private static string GetEffectiveGlobalPackagesFolder(
+            string packagesDirectoryParameter,
+            string solutionDirectoryParameter,
+            PackageRestoreInputs packageRestoreInputs,
+            string globalPackagesFolder)
+        {
+            if (!string.IsNullOrEmpty(packagesDirectoryParameter))
```

Comment:
0. Use the parameter
1. if packages folder is rooted the return it.
2. ifs to compute the solution directory
3. If there is a solution directory, combine it with the packages folder (assuming it is not null)
4. If nothing was figured out throw